packages/webdriverio/src/utils/interception/devtools.ts

```diff
@@ -220,10 +224,27 @@ export default class DevtoolsInterception extends Interception {
     /**
      * Does everything that `mock.clear()` does, and also
      * removes any mocked return values or implementations.
+     * Restored mock does not emit events and could not mock responses
      */
-    restore () {
+    async restore (sessionMocks = SESSION_MOCKS, cdpSessions = CDP_SESSIONS) {
         this.clear()
         this.respondOverwrites = []
+        this.restored = true
+        const handle = await this.browser.getWindowHandle()
+
+        log.trace(`Restoring mock for ${handle}`)
+        sessionMocks[handle].delete(this)
+
+        if (sessionMocks[handle].size) {
+            return
+        }
+
+        log.trace(`Disabling fetch domain for ${handle}`)
+        return cdpSessions[handle].send('Fetch.disable')
```

Comment:
This would restore mocking for all mocks right? Can you verify that you can create two mocks within the same session and have only one restored?

Reply:
Yup. Fixed in https://github.com/webdriverio/webdriverio/pull/10413/commits/ee4f27650c015bb7ddc2a8a1eff56542fa2edcb3

Also changed type [here](https://github.com/webdriverio/webdriverio/pull/10413/commits/ee4f27650c015bb7ddc2a8a1eff56542fa2edcb3#diff-0897b632afeab5d0ee660ee9e1ad90fade9350a431d9090571e4c934a7ea8ec9L14-R14)
and [used await instead of return here](https://github.com/webdriverio/webdriverio/pull/10413/commits/ee4f27650c015bb7ddc2a8a1eff56542fa2edcb3#diff-81b3fe6cd1e82bc9b5323f327cb91b6fba1a061ab6509f3b451676517a1ea598L50-R51), because `Promise<Promise<void>>` and `Promise<void>` are not the same thing